internal/dataplane/sendconfig/sendconfig.go

```diff
@@ -288,3 +269,60 @@ func isConflictErr(err error) bool {
 
 	return false
 }
+
+type KonnectAwareClient interface {
+	IsKonnect() bool
+}
+
+type StatusClient interface {
+	Status(context.Context) (*kong.Status, error)
+}
+
+// hasConfigurationChanged verifies whether configuration has changed by comparing old and new config's SHAs.
+// In case the SHAs are equal, it still can return true if a client is considered crashed based on its status.
+func hasConfigurationChanged(
+	ctx context.Context,
+	oldSHA, newSHA []byte,
+	client KonnectAwareClient,
+	statusClient StatusClient,
+	log logrus.FieldLogger,
+) (bool, error) {
+	if !bytes.Equal(oldSHA, newSHA) {
+		return true, nil
+	}
+	if !hasSHAUpdateAlreadyBeenReported(newSHA) {
+		log.Debugf("sha %s has been reported", hex.EncodeToString(newSHA))
+	}
+	if !client.IsKonnect() {
+		crashed, err := hasKongCrashed(ctx, statusClient, log)
+		if err != nil {
+			return false, fmt.Errorf("failed to verify kong readiness: %w", err)
+		}
+		// Kong instance has crashed, we should push config despite the oldSHA and newSHA being equal.
+		if crashed {
+			return true, nil
+		}
+	}
+
+	return false, nil
```

Comment:
This I assume it branch for only Konnect clients. Can you add some comments in here w.r.t why we always return `false` here?